client/src/partials/budget/analysis/analysis_budget.js

```diff
@@ -0,0 +1,420 @@
+        if(account.classe === 6 || account.classe === 2){
+          if(account.budget > account.balance) {
+            account.surplus = account.budget - account.balance;
+            account.deficit = 0;
+          } else if (account.budget < account.balance) {
+            account.deficit = account.balance - account.budget;
+            account.surplus = 0;
+          } else {
+            account.deficit = 0;
+            account.surplus = 0;
+          }
+        } else if(account.classe === 7 || account.classe === 1 || account.classe === 5){
+          if(account.budget < account.balance) {
+            account.surplus = account.balance - account.budget ;
+            account.deficit = 0;
+          } else if (account.budget > account.balance) {
+            account.deficit = account.budget - account.balance;
+            account.surplus = 0;
+          } else {
+            account.deficit = 0;
+            account.surplus = 0;
+          }
+        }
+        totalSurplus += account.surplus;
+        totalDeficit += account.deficit;  
+
+        parent = accountModel.get(account.parent);
+        depth = 0;
+        while (parent) {
+          depth += 1;
+          parent = accountModel.get(parent.parent);
+        }
+        account.depth = depth;
+      });
+      $scope.total.surplus = precision.round(totalSurplus);
+      $scope.total.deficit = precision.round(totalDeficit);  
+    }
+
+    function start(models) {
+      angular.extend($scope, models);
+      $scope.accounts.data.forEach(function (acct) {
+        if ((acct.type !== 'title') && (acct.balance === null)) {
+          acct.balance = 0.0;
+        }
+      });
+
+      filterAccounts($scope.accounts.data);
+      addBudgetData();
+      parseAccountDepth($scope.accounts.data, $scope.accounts);
+      session.mode = 'budget-analysis';
+    }
+
+    function displayAccounts() {
+      dependencies.accounts.query = '/InExAccounts/' +enterprise_id;
+      // Process period
+      var periodCriteria;
+      var selectedPeriod = session.periods.filter(function (p) {
+        return p.id === config.period_id;
+      })[0];
+      if (selectedPeriod.period_number === 0) {
+        dependencies.budgets.query.where = ['period.fiscal_year_id=' + config.fiscal_year_id];
+      } else {
+        dependencies.budgets.query.where = ['period.fiscal_year_id=' + config.fiscal_year_id, 'AND', 'period.id='+selectedPeriod.id];
+      }
+      validate.refresh(dependencies, ['accounts', 'budgets'])
+      .then(start);
+    }
+
+    function previousFYBudget(selectedPreviousFY) {
+      // Get budget data for all previous fiscal years
+      $scope.fiscalYearBudget = {};
+      for(var fy in selectedPreviousFY) {
+        var budget = { 
+          id           : selectedPreviousFY[fy].id, 
+          data         : null, 
+          totalBudget  : 0, 
+          totalBalance : 0 
+        };
+        $scope.fiscalYearBudget[budget.id] = {};
+        getBudget(budget.id);
```

Comment:
The function name `getBudget(id)` sounds like it should get _and return_ a budget.  In other words, I would expect you to use it like this:
```
var budget = getBudget(budget.id);
```

If the function does not return anything, it isn't really 'getting' anything, is it?  Perhaps it needs a better name?  Or maybe to be refactored into several functions, one that gets a budget, another that does performs some logic on that budget?